### PR TITLE
8323159: Consider adding some text re. memory zeroing in Arena::allocate

### PR DESCRIPTION
**Pull request by minborg**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8323526](https://bugs.openjdk.org/browse/JDK-8323526) to be approved

### Issues
 * [JDK-8323159](https://bugs.openjdk.org/browse/JDK-8323159): Consider adding some text re. memory zeroing in Arena::allocate (**Bug** - P4)
 * [JDK-8323745](https://bugs.openjdk.org/browse/JDK-8323745): Missing comma in copyright header in TestScope (**Bug** - P1)
 * [JDK-8323526](https://bugs.openjdk.org/browse/JDK-8323526): Consider adding some text re. memory zeroing in Arena::allocate (**CSR**)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk22.git pull/77/head:pull/77` \
`$ git checkout pull/77`

Update a local copy of the PR: \
`$ git checkout pull/77` \
`$ git pull https://git.openjdk.org/jdk22.git pull/77/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 77`

View PR using the GUI difftool: \
`$ git pr show -t 77`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk22/pull/77.diff">https://git.openjdk.org/jdk22/pull/77.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk22/pull/77#issuecomment-1892463686)

### CODE REVIEW

File: test/jdk/java/foreign/TestScope.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2024 Oracle and/or its affiliates. All rights reserved.
```
Comment (AlanBateman) [inline test/jdk/java/foreign/TestScope.java]: Can you hold off on this until TestScope.java's copyright header is fixed in main line.

Comment (minborg) [reply test/jdk/java/foreign/TestScope.java]: This has been fixed now.

**Review by mcimadamore — APPROVED** [review]
Looks good

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**